Frontend

josepabellana opened the pull request.

MarijaVitkauskaite committed:
feat: add explanations for bubble, selection and insertion

IP1llar committed:
feat: implement minions return to base after sorting

vadim-nest committed:
style(style left bar game page): style left bar game page

MarijaVitkauskaite committed:
style: style buttons on hover yellow!! LOVE IT

MarijaVitkauskaite committed:
feat: create scroll to learn more button

majilaii committed:
feat(sortinggignginewoinroiqwnwq): merge algo and also added MORE CUSTOMIZATION

width, delay, max_num val for new arr, padding, margin, and also added merge sort, and also fixed
visualize new array bugs

vadim-nest committed:
style: add some styling

majilaii committed:
feat(merge sort): implemented ASC/DESC mode

majilaii committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

MarijaVitkauskaite committed:
feat: make profile responsive until 950

IP1llar committed:
feat: begin implementing game stats

josepabellana committed:
feat(maze prepared to work with the backend logic): api calls prepared

MarijaVitkauskaite committed:
style: update profile component with the responsivness

MarijaVitkauskaite committed:
fix: update profile component

vadim-nest committed:
style: add shop component

IP1llar committed:
fix(game coins): update previous state when adding new towers

josepabellana commented on Client/src/components/learning/curveSortSVG.tsx:
```diff
@@ -0,0 +1,145 @@
+
+            <image className='decorations' x="-2" y="9" href={tallGrassPic} height='5' width='5' />
+            <image className='decorations' x="-7" y="35" href={tallGrassPic} height='5' width='5' />
+            <image className='decorations' x="33" y="3" href={tallGrassPic} height='5' width='5' />
```
Clean code please

josepabellana commented on Client/src/components/leftBar.tsx:
```diff
@@ -0,0 +1,153 @@
+    })
+  }
+
+  function zoomOut(amount: number) {
```
Would be good if we extract all the functionalities on a functionalities foldes

josepabellana commented on Client/src/components/leftBar.tsx:
```diff
@@ -0,0 +1,153 @@
+          <div className='zoomInButton' onMouseEnter={() => zoomHover('in', 'var(--yellow)')} onMouseLeave={() => zoomHover('in', 'var(--white-green)')} onClick={() => zoomIn(10)}>
+            <ZoomInSVG />
+          </div>
+          <div className='zoomOutButton' onMouseEnter={() => zoomHover('out', 'var(--yellow)')} onMouseLeave={() => zoomHover('out', 'var(--white-green)')} onClick={() => zoomOut(10)}>
```
Well done with the zooms

josepabellana commented on Client/src/css/curveSortSVG.css:
```diff
@@ -0,0 +1,89 @@
+  .path-labels {
+    width: fit-content;
+    height: fit-content;
+    position: absolute;
```
clean code please

josepabellana commented on Client/src/utils/sorting-helper-visual.ts:
```diff
@@ -0,0 +1,168 @@
+  //     const bar = arrayBars[i] as HTMLElement 
+  //     await delay(50)
+  //     bar.style.background = 'var(--main-green)'
+  // }
```
Clean code

IP1llar commented on Client/src/components/sortingLessons/bubbleLesson.tsx:
```diff
@@ -0,0 +1,86 @@
+  }, []);
+
+  useEffect(() => {
+    console.log(array);
```
Not needed

josepabellana replied:
Solved!

IP1llar commented on Client/src/components/sortingLessons/visualization.tsx:
```diff
@@ -0,0 +1,50 @@
+import '../../css/sorting.css';
+
+
+export default function Visualization(props: any) {
```
Can this be changed from any to the actual types

IP1llar commented on Client/src/components/game.tsx:
```diff
@@ -1,56 +1,216 @@
 import { bFS, dijkstra, getDirection, vBFS } from '../utils/path-finding-algo';
+import { bubbleSortAlgo } from '../utils/sorting-algo';
 
 function Game() { // TODO: Extract logic to maze class
```
Separate into different files

IP1llar commented on Client/src/components/leftBar.tsx:
```diff
@@ -0,0 +1,153 @@
+import FlagSVG from './svg/flagSVG';
+import MediaQuery from 'react-responsive';
+import { useAppSelector } from '../features/hooks';
+// import zoomInSVG from '../assets/game-different/search-3079.svg'
```
Remove comments

IP1llar commented on Client/src/components/maze.tsx:
```diff
@@ -31,56 +40,81 @@ function Maze({boxSize, setMazeCompleted, setCurrentMinion, minions, setCurrentT
-      })
-    }
-    else if (allTilesHidden) {
+    // async function mazeInit(){
```
Can this function be moved to a new file?

IP1llar commented on Client/src/components/minion.tsx:
```diff
@@ -22,9 +25,16 @@ function Minion({boxSize, minion, setCurrentMinion, setCurrentTile}: {
-      </div>
+    <div onClick={handleClick} onContextMenu={handleContextMenu}  id={`${minion.id}`} className={`minion ${minion.rotation} ${(minion.inTower && !minion.rotation)&& 'minionInTower'}`} style={{fill: `red`, height: `${boxSize}px`, width: `${boxSize}px`, top: `${boxSize*minion.yPos}px`, left: `${boxSize*minion.xPos}px`}}>
+    {
+      minion.type === 'Squirrel' ? <Squirrel currentPlayer={`${minion.alignment}-color`} /> :
```
Can this be a function?

vadim-nest replied:
Done

IP1llar committed:
feat(game, path-finding-algo): implement efficient visual DFS

vadim-nest committed:
style: minor changes to the shop component

vadim-nest committed:
style: some progress with the right panel in the game

MarijaVitkauskaite committed:
feat: new nut component for learning page implemented

MarijaVitkauskaite committed:
feat: components for SVGs for learning page

IP1llar committed:
feat(game, path-algos): implement efficient visual Dijkstra

vadim-nest committed:
style: add style to the shop

IP1llar committed:
feat(left and right bars): implement unique names for each minion

majilaii committed:
feat(sorting redux backend): finnished quick sort redux backend start

IP1llar committed:
feat(path algs): implement a* path alg

vadim-nest committed:
feat: give different stats to different animals

vadim-nest committed:
style: design the shop page

majilaii committed:
feat(sorting curvesvg models backend router): sorting ASC mode nuts dsiabled/done logic

majilaii committed:
fix(sort svg): also implemented dynamic profile page keep tracking of sort learn progress

majilaii committed:
fix(svg bug fi): svg bug for bubble sort

majilaii committed:
fix(nvm had to fix it again for bubble): bubble svg bug

majilaii committed:
feat(css profile): fixed css for profile pic

IP1llar committed:
feat(sorting algo in maze): implement sorting algorithms in the maze

vadim-nest committed:
feat: add invaded towers count. Finish styling the shop

IP1llar committed:
fix: prevent animations from starting more than once for a sigle sorting

vadim-nest committed:
feat: design the shop bar. Add home base SVGs

majilaii committed:
Update README.md

majilaii committed:
Merge pull request #7 from IP1llar/majilaii-patch-1

Update README.md

majilaii committed:
Update README.md

IP1llar committed:
Revert "feat: design the shop bar. Add home base SVGs"

This reverts commit 085fe72202058d5d3921fe86aedfed4e56709de3.

IP1llar committed:
Update README.md

IP1llar committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

vadim-nest committed:
feat: add an arrow button to the shop

IP1llar committed:
feat: close tower popup on zoom

IP1llar committed:
fix: undo deleting a commit

majilaii committed:
feat(a lot of things): css for profile redux path find

IP1llar committed:
feat: add home svg for each player

IP1llar committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

vadim-nest committed:
feat: add a new close store button

josepabellana committed:
feat(add dfs learning page): add dfslearning page

josepabellana committed:
feat(add bfs page and algo): add bfs algo

josepabellana committed:
feat(add dijkstra lesson): add dynamic display logic

josepabellana committed:
style(added links to learning path): add links &behavior test

MarijaVitkauskaite committed:
fix: create coinSVG and use it in profile

IP1llar committed:
feat: path animates only for selected minion

IP1llar committed:
feat: speed up path animation

majilaii committed:
feat(buttons sorting fuck heap): fuck heap

majilaii committed:
feat(forgot to push this one): qwoienwqoi

IP1llar committed:
feat: reset path animation when switching current Minion

josepabellana committed:
feat(add layout to path finding pages): refactor path finding pages

josepabellana committed:
style(cleaning): cleaning

vadim-nest committed:
feat: larger animals

MarijaVitkauskaite committed:
feat: implement socket.io

majilaii committed:
feat(final sorting adjustments): qwe

vadim-nest committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

josepabellana committed:
feat(delete avatars login in backend): moved avatars to front end

Added cypress instance

MarijaVitkauskaite committed:
feat: add socket.io server logic

MarijaVitkauskaite committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

MarijaVitkauskaite committed:
style: change learning path finding algorithm colors to make it more visible

josepabellana committed:
style(username responsive, stats text showing same font size): minor changes on design progile page

josepabellana committed:
style(git user export deleted by error): solved bug

IP1llar committed:
feat: implement socket io for creating and moving minions

josepabellana committed:
refactor(change user schema and add game schema): add game schema

MarijaVitkauskaite committed:
fix: fix buttons, now clickable (in learning path finding)

IP1llar committed:
fix(minion towers): minions enter towers through socketio

vadim-nest committed:
feat: start with the waiting room page

vadim-nest committed:
feat: start working on waitingRoom

IP1llar committed:
fix: remove some logs

vadim-nest committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

IP1llar committed:
feat: server generates a maze on start and delivers to all clients

IP1llar committed:
feat: generate tower number in server and players can only pick their own minions

IP1llar committed:
feat(gameover): implement game over popup

josepabellana committed:
test(cypress login with test user): testing Auth0 login

IP1llar committed:
fix(gameover): ensure win/lose displays correctly for the correct player

vadim-nest committed:
feat(design the waiting room): design the waiting room

josepabellana committed:
test(testing mazegame): e2E test maze

MarijaVitkauskaite committed:
feat: create socket.io for create and join room in the waiting room

vadim-nest committed:
feat: add routing to the navbar

vadim-nest committed:
feat: add routing to the navbar

majilaii committed:
fix(fixed css in learning page): qwewq

majilaii committed:
feat(all sorts play): some things dont work

majilaii committed:
fix(prevent all sorts from interfering with learning): qe

IP1llar committed:
test(client): configure jest in the client

IP1llar committed:
test(testing): implement bubble and insertion sort tests

josepabellana committed:
feat(changes on profile): profile changes

josepabellana committed:
fix(profile bug): profile bug

majilaii committed:
feat(i cant fking fix customize): qwe

majilaii committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

vadim-nest committed:
feat: add style to the waitingRoom page

josepabellana committed:
feat(create algorithm hook): hook

josepabellana committed:
style(minor change): minor change

majilaii committed:
feat(fixed fucking sort and updated hook): qwe

majilaii committed:
feat(sort hook): sort hook

josepabellana committed:
style(adding options to algo hook): algo hook

MarijaVitkauskaite committed:
refactor: clean up

josepabellana committed:
style(add astar page): adding page

josepabellana committed:
fix(main changes): add page

vadim-nest committed:
feat: change style waiting room

vadim-nest committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

vadim-nest committed:
fix: fix the spacing in the waiting room page

IP1llar committed:
feat(waiting room game): integrate waiting room with the game

IP1llar committed:
feat(socket io): only emit minion movement and creation to the room

IP1llar committed:
feat(socket io): only emit entering tower to the room

IP1llar committed:
feat: generate a new maze for each room

MarijaVitkauskaite committed:
feat: local storage implemented

MarijaVitkauskaite committed:
feat: local storage implemented

IP1llar committed:
fix: uninstall moment.js

MarijaVitkauskaite committed:
feat: local storage

IP1llar committed:
fix(localstorage): initialise localstorage variables if they are null

vadim-nest committed:
feat: add some style to the gameOver window

majilaii committed:
feat: added customization sort

majilaii committed:
feat(input ref at profile): qwe

josepabellana committed:
refactor(changing avatar to auth0 picture): avatar change

josepabellana committed:
refactor(minor fix): minor fix

majilaii committed:
feat(fixed socket bug): qopiepw

josepabellana committed:
fix(fixing bug): small bug

majilaii committed:
fix(socket): sockt fix

majilaii committed:
feat(changed it so that when u go back from gam eover u go back to game waiting room): qwe

majilaii committed:
fix(socket disconnect fixes): socket

josepabellana committed:
refactor(add maze with weights): starting astar visualization

josepabellana committed:
style(deleting avatar values): deleting avatar type on user models

IP1llar committed:
fix(socket): prevent player from joining their own game

MarijaVitkauskaite committed:
refactor(navbar.tsx): refactor onClick function

MarijaVitkauskaite committed:
refactor: refactor navbar.tsx, delete login

MarijaVitkauskaite commented on Client/src/__tests__/sorting-Algorithms/insertion.tsx:
```diff
@@ -0,0 +1,29 @@
+  test('Sorts an array', () => {
+    let array = [5,3,6,7,8,3];
+    let sortedAnimations = insertionSortAlgo([...array], true);
+    console.log(sortedAnimations);
```
Remove console logs after finishing with the task

vadim-nest committed:
style: tidy up the code of the shop.tsx

MarijaVitkauskaite commented on Client/src/components/leftBar.tsx:
```diff
@@ -0,0 +1,252 @@
+    );
+  }
+
+  // Working, where do I put it?
+  // (document.getElementById('text-reverse-yep') as unknown as HTMLElement).textContent = `${gameStats.p2Towers.length}`;
+
```
Complete or delete

MarijaVitkauskaite commented on Client/src/components/leftBar.tsx:
```diff
@@ -0,0 +1,252 @@
+            textReverse="text-reverse-yep"
+          />
+        </div>
+        {/* <h3 className='score-you'>3</h3>
+        <h3 className='score-opponent'>4</h3> */}
```
Delete!

MarijaVitkauskaite commented on Client/src/components/leftBar.tsx:
```diff
@@ -0,0 +1,252 @@
+        <div className="scores"></div>
+      </MediaQuery>
+      <div className="selected-info">
+        {/* {currentMinion !== null && <h1 className='current-minion-name-left-bar'>{minions[currentMinion].id}</h1>} */}
+
```
...

MarijaVitkauskaite commented on Client/src/components/leftBar.tsx:
```diff
@@ -0,0 +1,252 @@
+                }
+              </h1>
+            )}
+            {/* <h1 className='current-tower-name-left-bar'>{currentTower.alignment}</h1> */}
+
```
...

MarijaVitkauskaite commented on Client/src/components/maze.tsx:
```diff
@@ -31,56 +42,84 @@ function Maze({boxSize, setMazeCompleted, setCurrentMinion, minions, setCurrentT
-        }
+      if (mazeGenerated === false) {
+        setMazeGenerated(true);
+        // const {graph, visited, classes, towers} = generateMaze(width,height)
```
...

MarijaVitkauskaite commented on Client/src/components/maze.tsx:
```diff
@@ -31,56 +42,84 @@ function Maze({boxSize, setMazeCompleted, setCurrentMinion, minions, setCurrentT
+            xPos: tower[0]%width,
+            yPos: Math.floor(tower[0]/width),
+            numbers: tower[1],
+            color: 'red',
```
Use global variable colours: var(--red)

MarijaVitkauskaite commented on Client/src/components/maze.tsx:
```diff
@@ -31,56 +42,84 @@ function Maze({boxSize, setMazeCompleted, setCurrentMinion, minions, setCurrentT
+      }
+      else if (allTilesHidden && currentGraph) {
+        const mazeTiles = document.getElementsByClassName('mazeTile');
+        console.log({mazeTiles})
```
Delete console logs

MarijaVitkauskaite commented on Client/src/components/maze.tsx:
```diff
@@ -31,56 +42,84 @@ function Maze({boxSize, setMazeCompleted, setCurrentMinion, minions, setCurrentT
   },[maze]);
 
   return (
     <>
```
This is already in a div, do we need an empty tag as well?

IP1llar replied:
No we don't

MarijaVitkauskaite commented on Client/src/components/profile/profile.tsx:
```diff
@@ -0,0 +1,180 @@
+            </h4>
+          </div>
+          <h3 className="created">
+            Account created {new Date().toLocaleDateString()}
```
Change date to when the account was created, not current one (ask Josep about the data from Auth0)

MarijaVitkauskaite commented on Client/src/components/rightBar.tsx:
```diff
@@ -0,0 +1,136 @@
+function crossButtonHover(isOnHover: boolean) {
+  document.querySelectorAll('.close-cross-green').forEach((svgEl) => {
+    isOnHover
+      ? ((svgEl as unknown as HTMLElement).style.fill = `#96B0A2`)
```
What is this colour? If you're using it, please add it to the global variables to keep consistency

vadim-nest replied:
Done

MarijaVitkauskaite commented on Client/src/components/rightBar.tsx:
```diff
@@ -0,0 +1,136 @@
+  currentPlayer: 'p1' | 'p2'
+}) {
+ 
+  const [shopOpen, setShopOpen] = useState(false);
```
Keep your states at the top of the function, for readability

vadim-nest replied:
Done

MarijaVitkauskaite commented on Client/src/components/sortingLessons/allSortsPlay.tsx:
```diff
@@ -0,0 +1,195 @@
+  useEffect(() => {
+    algoChosen(choiceOfAlgo)
+
+    return(() => console.log('stop listening'))
```
Is it needed?

MarijaVitkauskaite commented on Client/src/components/sortingLessons/mergeSort.tsx:
```diff
@@ -0,0 +1,86 @@
+
+      <div className="lesson-wrapper-2">
+        <div>
+        {!clicked && 
+           < button className="button clickSort" onClick={() => initArr()}>
+            new array
+          </button>}
+
+          {!clicked &&!isSorted &&
+            <button
+              className="button clickSort"
+              onClick={() => {
+                setClicked(true);
+              }}
+            >
+              visualize
+            </button>
+          }
+
+
+          {!isSorted && clicked &&
+           <button 
+           className="button clickSort">
+             wait... </button>
+          }
+
+        </div>
```
Can this be a component? It repeats in every algorithm

MarijaVitkauskaite commented on Client/src/components/waitingRoom.tsx:
```diff
@@ -0,0 +1,207 @@
+
+  useEffect(() => {
+    socket.on('Game start', () => {
+      console.log('Game started');
+      console.log(store.getState().game)
```
Remove after completion

MarijaVitkauskaite commented on Client/src/components/waitingRoom.tsx:
```diff
@@ -0,0 +1,207 @@
+              </h1>
+            </button>
+            {createClicked ? (
+              // <p>
+              //   {id}
+              // </p>
```
Delete if not used

IP1llar replied:
deleted

MarijaVitkauskaite commented on Client/src/components/waitingRoom.tsx:
```diff
@@ -0,0 +1,207 @@
+                    name="fname"
+                    placeholder="type the code here"
+                  />
+                  <button className="submit-the-code-button">✓</button>
```
close in one tag if no content inside

IP1llar replied:
There's a little tick

MarijaVitkauskaite commented on Client/src/css/aStar-lesson.css:
```diff
@@ -0,0 +1,57 @@
+.whole-page-wrapper {
+    background-color: var(--white);
+    height: 90vh;
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: flex-start;
+    overflow: scroll;
+  }
```
Repeating CSS again, this class is in BFS CSS file

MarijaVitkauskaite commented on Client/src/css/allSortsPlay.css:
```diff
@@ -0,0 +1,79 @@
+    font-weight: bolder;
+}
+
+.disabled{
+    color: gray;
+}
+
+.disabled:hover{
+    cursor: default;
+    color: grey;
+}
```
If you're using grey, especially more than once, add it to the global variables

vadim-nest replied:
Done

MarijaVitkauskaite commented on Client/src/css/allSortsPlay.css:
```diff
@@ -0,0 +1,79 @@
+
+
+.nice-select {
+    background-color: #fff;
```
var(--white)

vadim-nest replied:
Done

MarijaVitkauskaite commented on Client/src/css/allSortsPlay.css:
```diff
@@ -0,0 +1,79 @@
+.nice-select {
+    background-color: #fff;
+    border-radius: 5px;
+    border: solid 1px #e8e8e8;
```
Global variables

vadim-nest replied:
Done

MarijaVitkauskaite commented on Client/src/css/allSortsPlay.css:
```diff
@@ -0,0 +1,79 @@
+    transition: all 0.2s ease-in-out;
+    width: auto; }
+.nice-select:hover {
+      border-color: #dbdbdb; }
+.nice-select:active, .nice-select.open, .nice-select:focus {
+      border-color: #999; }
```
Global variables

vadim-nest replied:
Done

MarijaVitkauskaite commented on Client/src/css/curveSortSVG.css:
```diff
@@ -0,0 +1,74 @@
+  fill: var(--yellow);
+
+}
+
+/* .circles-after {
+  fill: var(--dark-brown);
+}
+
+.nut-after {
+  fill: var(--yellow);
+} */
+
```
Delete

MarijaVitkauskaite commented on Client/src/css/leftBar.css:
```diff
@@ -0,0 +1,186 @@
+  margin: auto 10px auto 10px;
+}
+
+/* .left-just-two-columns {
+  display: flex;
+} */
+
+/* .left-just-top {
+  display: inline;
+} */
+
```
Delete

MarijaVitkauskaite commented on Client/src/css/shop.css:
```diff
@@ -0,0 +1,93 @@
+  margin-bottom: 0px;
+  padding-bottom: 0px;
+  white-space: nowrap;
+  /* overflow: hidden; */
+  /* text-overflow: ellipsis; */
```
Delete

vadim-nest replied:
Done

MarijaVitkauskaite commented on Client/src/css/tower.css:
```diff
@@ -0,0 +1,4 @@
+.tower {
+  position: absolute;
+  z-index: 9;
```
Are we already using 8 z-indexes?

MarijaVitkauskaite commented on Client/src/css/waitingRoom.css:
```diff
@@ -0,0 +1,188 @@
+  margin-left: auto;
+}
+
+.waiting-page-join-button {
+  /* margin-left: 0; */
+  /* align-self: flex-start; */
+}
+
+.right-side-selection {
+  max-width: 40%;
+  display: flex;
+  flex-direction: column;
+  padding-left: 3vw;
+}
+
+.wr-play-button {
+  /* margin-left: -40px; */
+  /* margin-right: auto; */
+}
+
```
Delete comments

vadim-nest replied:
Done

MarijaVitkauskaite reviewed (commented):
Good job, everyone! ❤️ 
Main themes about keeping the code clean, readable and DRY: 
- format your files with prettier (if you don't like how it looks, format it after you're done) so it's easy to read your code
- delete comments and console logs

MarijaVitkauskaite committed:
feat: make learning page responsive

MarijaVitkauskaite committed:
feat(welcome.tsx): change page design to make responsive

IP1llar committed:
fix(socket io): search now works better

IP1llar committed:
fix(socket): create and join now work

majilaii committed:
fix(seesm to have fixed the go back/then search bug): qe

MarijaVitkauskaite committed:
feat(game navbars): change game navbars to be responsive until 950px

MarijaVitkauskaite committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

IP1llar committed:
feat(game socketio): implement server keeping track of game state

vadim-nest committed:
feat: small design changes, lesson page

vadim-nest committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

IP1llar committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

vadim-nest committed:
fix: fix the styling of algorithms in the game

vadim-nest committed:
fix: fix the bug with small zoom buttons

josepabellana committed:
feat(refactoring path finding visuals): refactoring visual path

josepabellana committed:
fix(maze conflict): maze conflict

IP1llar committed:
feat(socket io game state): server keeps track of coins

MarijaVitkauskaite committed:
feat: start left bar modal, create new file: LeftBarSmall

IP1llar committed:
fix: fix zoom

IP1llar committed:
fix(socket io): prevent host/join and play logic messing each other up

IP1llar committed:
feat(socketio profile): implement storing of game in database

josepabellana committed:
feat(new learning  path): new learning path

josepabellana committed:
style(change learning path styles): styling learnigng path

josepabellana committed:
style(minor changes in style): style

IP1llar committed:
chore(game): move game files to their own folder

IP1llar committed:
feat(game redux): implement redux for some game state

zoomIn, zoomOut, mazeComplete, setWaitingForTile, setCurrentGraph, setAllTilesHidden,
updateCurrentMinion, updateCurrentTower

josepabellana commented on Client/package.json:
```diff
@@ -6,25 +6,47 @@
+    "@babel/preset-env": "^7.20.2",
+    "@babel/preset-react": "^7.18.6",
+    "@babel/preset-typescript": "^7.18.6",
+    "@testing-library/jest-dom": "^5.16.5",
```
Can anyone explain me what babel is used for? Genuine question

IP1llar committed:
feat(redux game): move gameStats and finalGameStats to redux

josepabellana commented on Client/src/__tests__/sorting-Algorithms/bubble.tsx:
```diff
@@ -0,0 +1,30 @@
+  test('Sorts an array', () => {
+    let array = [5,3,6,7,8,3];
+    let sortedAnimations = bubbleSortAlgo([...array], true);
+    expect(sortedAnimations).toEqual(
```
We can use the hook here useAlgo('namealgo',[array],reverersed(true or false))

IP1llar committed:
feat(game redux): move zoomed to redux

IP1llar committed:
feat(game redux): move minions to redux

IP1llar committed:
feat(game redux): move currentTile to redux

IP1llar committed:
feat(game redux): move movingMinions to redux

majilaii committed:
fix(fixed joseps bug): qwe

IP1llar committed:
feat(game redux): move maze to redux

IP1llar committed:
feat(game redux): move mazeGenerated to redux and clean up

josepabellana committed:
refactor(check merge in dfs): check merge

josepabellana committed:
fix(fix bug with new visual): bug fix

majilaii committed:
fix(half fix for renewing array path finding): qwe

IP1llar committed:
feat(game redux): default state

IP1llar commented on Client/src/components/learning/learning.tsx:
```diff
@@ -0,0 +1,96 @@
+import MediaQuery from 'react-responsive';
+
+export default function LearningPage() {
+  const { getAccessTokenSilently, isAuthenticated } = useAuth0();
```
Unused so can be removed

MarijaVitkauskaite replied:
Done

josepabellana replied:
cleaned

IP1llar commented on Client/src/components/learning/learning.tsx:
```diff
@@ -0,0 +1,96 @@
+        </div>
+        <button
+          className="scroll-learning"
+          onClick={() => window.scrollBy(0, window.innerHeight * 0.9)}
```
Nice scrolling

MarijaVitkauskaite replied:
Will make it scroll to an element

josepabellana replied:
the next two states werent used too...

IP1llar commented on Client/src/components/learning/learning.tsx:
```diff
@@ -0,0 +1,96 @@
+                others.
+              </p>
+            </div>
+            <MediaQuery minWidth={601}>
```
Can we use 'useMediaQuery' here to make this a bit cleaner?

MarijaVitkauskaite replied:
Will look into it

IP1llar commented on Client/src/components/profile/profile.tsx:
```diff
@@ -0,0 +1,192 @@
+  let currentSortProgress = 0;
+  let currentPathProgress = 0;
+
+  for (let bool of userRedux.sortLessons) {
```
A more descriptive name than bool?

vadim-nest replied:
Done

IP1llar commented on Client/src/components/sortingLessons/allSortsPlay.tsx:
```diff
@@ -0,0 +1,226 @@
+    setArray(generateArray(20, MIN_VAL, MAX_VAL));
+  }, []);
+
+  function algoChosen(choice: string) {
```
Can we use the new hook for this?

IP1llar commented on Client/src/components/sortingLessons/bubbleLesson.tsx:
```diff
@@ -0,0 +1,112 @@
+        <div className="pseudoBlock">
+        <h1 className='lesson-h1'>{paragraphs.sortName}</h1>
+          <code>
+          <span className='code indent0 '> bubbleSort(array) </span> <br />
```
This looks promising

josepabellana replied:
xd, cant guess who push this

IP1llar commented on Client/src/components/sortingLessons/quickSort.tsx:
```diff
@@ -0,0 +1,86 @@
+  let paragraphs = {
+    sortName: 'Quick sort',
+    firstP:
+      'Bubble Sort is the simplest sorting algorithm that swaps two elements if they are in the wrong order. As we go through each element, if the current element is bigger than the next one, we swap them. This algorithm is not suitable for large data sets as its average and worst-case time complexity is quite high.',
```
This seems familiar

IP1llar commented on Client/src/components/sortingLessons/visualization.tsx:
```diff
@@ -0,0 +1,52 @@
+
+export default function Visualization(props: any) {
+
+    let {array, animations, clicked, sortingAlgo, margin, fontColor, fontSize, paddingTop, width, delay, height, tower, setClicked,setIsSorted} = props
```
Can we set the default values here?

IP1llar replied:
Yes we can

IP1llar commented on Client/src/components/svg/towerSVG.tsx:
```diff
@@ -0,0 +1,45 @@
+import '../../css/TowerSVG.css'
+export default function TowerSVG ({playerClass, playerClassShadow}: {playerClass: string, playerClassShadow: string}) {
```
I don't think we need playerClassShadow anymore

vadim-nest replied:
Gone

IP1llar commented on Client/src/components/auth0Provider.tsx:
```diff
@@ -12,10 +12,6 @@ const Auth0ProviderWithHistory = (props:any) => {
-
   return (
     <Auth0Provider
       domain={"dev-mujh303ammb4fy01.uk.auth0.com"} //TODO add to .env file(auth0 joseph's domain)
```
Yeah an .env file would be great

IP1llar commented on Client/src/components/graphVertex.tsx:
```diff
@@ -0,0 +1,27 @@
+
```
Does this file belong in pathfindingLessons folder?

josepabellana replied:
yes, should be part of the pathFinding Lessons folder

IP1llar commented on Client/src/components/navbar.tsx:
```diff
@@ -43,26 +53,27 @@ function Navbar() {
             aMAZEthing
           </button>
           <MediaQuery minWidth={951}>
```
Can the MediaQuery be replaced by useMediaQuery?

MarijaVitkauskaite replied:
Will look into it

IP1llar commented on Client/src/components/waitingRoom.tsx:
```diff
@@ -0,0 +1,231 @@
+import { useNavigate } from 'react-router-dom';
+import { useState, useEffect } from 'react';
+import socket from '../services/socket';
+import ProfileGameHistory from './profile/profileGameHistory';
+import { io } from 'socket.io-client';
```
No longer needed?

IP1llar replied:
Looking into it

IP1llar commented on Client/src/components/waitingRoom.tsx:
```diff
@@ -0,0 +1,231 @@
+    socket.off('receiveRoomId');
+    socket.off('receiveRoomIdHost');
+    socket.off('receiveRoomIdJoin');
+    socket.off('Game start');
```
Move to the return

IP1llar commented on Client/src/components/waitingRoom.tsx:
```diff
@@ -0,0 +1,231 @@
+      console.log('Received room ID', {roomId})
+      setId(roomId);
+      dispatch(updateRoomID(roomId));
+      dispatch(updatePlayer(player));
```
Can these be a single dispatch?

IP1llar replied:
Yes it can

IP1llar commented on Client/src/components/waitingRoom.tsx:
```diff
@@ -0,0 +1,231 @@
+    socket.on('Game start', () => {
+      const roomId = store.getState().game.roomId;
+      if (roomId) {
+        console.log('Game started', store.getState().game.roomId);
```
Remove these

IP1llar commented on Client/src/components/waitingRoom.tsx:
```diff
@@ -0,0 +1,231 @@
+      socket.emit(`ready${type}`, roomId);
+    });
+    socket.on('Game start', () => {
+      const roomId = store.getState().game.roomId;
```
Use useAppSelector

IP1llar commented on Client/src/components/waitingRoom.tsx:
```diff
@@ -0,0 +1,231 @@
+    });
+    return ()=>{ 
+      // console.log('clearing waiting');
+      // socket.emit('clear waiting', store.getState().game.roomId) // TODO: Currently this prevents them from joining the game on game start
```
Check if this is fixed

IP1llar commented on Client/src/components/waitingRoom.tsx:
```diff
@@ -0,0 +1,231 @@
+    if (!createClicked) {
+      // greyOutOtherButtons('Create');
+      console.log('playClicked');
+      setCreateClicked(true);
+      setJoinClicked(false);
+      setPlayClicked(false);
```
These can be joined into a single function (or dispatch if converted to redux)

vadim-nest replied:
Joined unto a single function

IP1llar commented on Client/src/components/waitingRoom.tsx:
```diff
@@ -0,0 +1,231 @@
+    }
+  }
+
+  // useEffect(() => {
```
Delete if not used

IP1llar commented on Client/src/components/welcome.tsx:
```diff
@@ -13,21 +10,22 @@ function Welcome() {
     <div className='welcome-page'>
       <MediaQuery minWidth={951}>
```
Use useMediaQuery with && instead?

IP1llar commented on Client/src/features/store.ts:
```diff
@@ -1,13 +1,38 @@
+import gameReducer from './game_slice';
 
- import userReducer from './user_slice'
+function saveToLocalStorage(store: any) {
```
Localstorage may be interferring/overwriting the game redux now so need to specify what is stored

MarijaVitkauskaite replied:
WIll fix it

IP1llar commented on Client/src/features/hooks.ts:
```diff
@@ -1,6 +1,37 @@
+
+
+
+export function useAlgo(algoName: string = 'bubble', array: any = [], reversed: boolean = false) {
```
Nice

IP1llar commented on Client/src/services/apiService.ts:
```diff
@@ -1,8 +1,13 @@
+
 const BASE_URL = 'http://localhost:3000'; //TODO Add to .env
 
 type ApiService = {
```
Where can this be used?

IP1llar commented on Client/src/services/socket.ts:
```diff
@@ -0,0 +1,7 @@
+
+const socket = io('http://localhost:3000');
+
+console.log('socket', socket);
```
Can be removed

IP1llar commented on Client/src/utils/sorting-helper-visual.ts:
```diff
@@ -0,0 +1,201 @@
+  //     console.log(bar)
+  //     await delay(50)
+  //     bar!.style.background = 'var(--dark-green)'
+  // }
```
Are we definitely deleting this?

IP1llar commented on Client/src/utils/sorting-helper-visual.ts:
```diff
@@ -0,0 +1,201 @@
+      await delay(DELAY_MS);
+      if (animations[i].length > 2) {
+        const [indexOne, elementOne, indexTwo, elementTwo] = animations[i];
+        document.getElementById(`${indexOne}arrayColumn${tower}`)!.style.backgroundColor = 'var(--red)';
```
If these are still causing trouble we can try useRef

IP1llar commented on Client/src/utils/types.ts:
```diff
@@ -1,16 +1,88 @@
+  sortingAlgo: 'bubble' | 'insertion' | 'selection' | 'merge' | 'quick';
+}
+
+export interface animal {
```
Need to add movementSpeed here and change it for each animal

IP1llar replied:
On it

IP1llar commented on Client/src/components/game/gameOver.tsx:
```diff
@@ -0,0 +1,68 @@
+
+  const {finalGameStats} = useAppSelector(state => state.game);
+
+  let win = finalGameStats.p1Towers.length > finalGameStats.p2Towers.length;
+  if (finalGameStats.p1Towers.length === finalGameStats.p2Towers.length) win = finalGameStats.p1Coins > finalGameStats.p2Coins;
+  if (currentPlayer === 'p2') win = !win;
```
Need to allow for a draw

IP1llar replied:
On it

IP1llar commented on Client/src/components/game/game.tsx:
```diff
@@ -0,0 +1,298 @@
+
+
+
+  useEffect(() => {
```
Can the socket logic be moved to another file?

IP1llar replied:
Not easily

IP1llar commented on Client/src/components/game/game.tsx:
```diff
@@ -0,0 +1,298 @@
+  useEffect(() => {
+    if (gameStats.timeRemaining === 0 && !gameEnded) {
+      dispatch(finalGameStats());
+      dispatch(updateGameEnded());
```
Can thesse be combined?

IP1llar commented on Client/src/components/game/game.tsx:
```diff
@@ -0,0 +1,298 @@
+    if(currentPlayer === player) {
+      socket.emit('new minion', type, roomId, player);
+    }
+    dispatch(addNewMinionState({type, player}));
+    if (currentPlayer === player) {
+      dispatch(updateCurrentMinion(newId));
+      dispatch(updateCurrentTower(null));
+    }
```
Combine into single dispatch?

IP1llar replied:
On it

IP1llar commented on Client/src/components/game/game.tsx:
```diff
@@ -0,0 +1,298 @@
+
+  function opponentMovement(direction: {xPos: number, yPos: number, rotation: 'minionU' | 'minionR' | 'minionL' | 'minionD' | '' }, minionId: number) {
+    dispatch(opponentMinionMovement({direction, minionId}));
+  }
```
Can we just use the dispatch instead of this function?

IP1llar commented on Client/src/components/game/game.tsx:
```diff
@@ -0,0 +1,298 @@
+    }
+    else {
+      directions = vBFS(comeFrom.xPos + comeFrom.yPos*width, goTo.xPos + goTo.yPos*width, currentGraph);
+    }
```
Can this be made into a hook?

IP1llar commented on Client/src/components/game/game.tsx:
```diff
@@ -0,0 +1,298 @@
+        if (currentMinion === minion.id) setPath(showPath, showThoughtProcess, minion.id); // TODO: get current Minion
+        requestAnimationFrame(step);
+      }
+      requestAnimationFrame(step);
```
Turn into setInterval and/or add to intervals array so it can be cleared

IP1llar commented on Client/src/components/game/game.tsx:
```diff
@@ -0,0 +1,298 @@
+    }
+  }, [currentMinion])
+
+  function setPath(path: number[], visited: number[], minionId: number | null) {
+    if (currentMinion === minionId) {
+      dispatch(updateMazePath({path, visited, minionId}));
+    }
+  }
```
Do we need this function anymore?

IP1llar commented on Client/src/components/game/game.tsx:
```diff
@@ -0,0 +1,298 @@
+  }
+
+  useEffect(() => { // TODO: Extract to a new file
+    if (waitingForTile && currentTile !== null && currentGraph && !movingMinions.includes(currentMinion as number)) {
```
Check we need all of these

IP1llar commented on Client/src/components/game/game.tsx:
```diff
@@ -0,0 +1,298 @@
+
+  async function enterTower(towerId: number, minionId: number, movedAfter = true) {
+
+    dispatch(minionEnterTower({minionId, towerId}));
+    if (currentMinion === minionId) {
+      dispatch(updateCurrentMinion(null));
+    };
```
Can be combined

IP1llar commented on Client/src/components/game/game.tsx:
```diff
@@ -0,0 +1,298 @@
+    const minion = minions[minionId];
+    const tower = towers.find(tower => tower.id === towerId) as TowerType;
+    let array = [...tower.numbers];
+    let animations = (
+      minion.sortingAlgo === 'bubble' ? bubbleSortAlgo([...tower.numbers], minion.alignment === 'p1') :
+      minion.sortingAlgo === 'insertion' ? insertionSortAlgo([...tower.numbers], minion.alignment === 'p1') :
+      minion.sortingAlgo === 'selection' ? selectionSortAlgo([...tower.numbers], minion.alignment === 'p1') :
+      minion.sortingAlgo === 'merge' ? mergeSortAlgo([...tower.numbers], minion.alignment === 'p1') :
+      quickSortAlgo(array, minion.alignment === 'p1')
+    )
```
Hook for this?

IP1llar commented on Client/src/components/game/game.tsx:
```diff
@@ -0,0 +1,298 @@
+      quickSortAlgo(array, minion.alignment === 'p1')
+    )
+    await new Promise((resolve, reject) => {
+      const interval = setInterval(()=>{
```
Need to stop this if player exits the game early

IP1llar commented on Client/src/components/game/game.tsx:
```diff
@@ -0,0 +1,298 @@
+        socket.emit('conquerTower', roomId, towerId, false, currentPlayer);
+      }
+    }
+    dispatch(minionExitTower({minionId, towerId}));
+    dispatch(addMovingMinion(currentMinion as number));
```
Combine?

IP1llar commented on Client/src/components/game/game.tsx:
```diff
@@ -0,0 +1,298 @@
+  function exitTower(towerId: number, minionId: number, movedAfter: boolean) {
+    let minion = minions[minionId];
+    let tower = towers.find(tower => tower.id === towerId) as TowerType;
+    if (minion.alignment === currentPlayer) {
+      if (tower.alignment === 'none') {
+        socket.emit('conquerTower', roomId, towerId, true, currentPlayer);
+      } else {
+        socket.emit('conquerTower', roomId, towerId, false, currentPlayer);
+      }
+    }
```
Make more succinct

IP1llar commented on Client/src/components/game/game.tsx:
```diff
@@ -0,0 +1,298 @@
+            currentMinion={currentMinion}
+            minions={minions}
+            currentPlayer = {currentPlayer}
+          />
```
Stop prop drilling and use new redux

vadim-nest replied:
On it..

vadim-nest replied:
Done

IP1llar commented on Client/src/components/game/gameOver.tsx:
```diff
@@ -0,0 +1,68 @@
+import '../../css/gameOver.css';
+import { useAppSelector } from '../../features/hooks';
+
+function GameOver ({currentPlayer} : {
```
useAppSelector

vadim-nest replied:
Done

IP1llar commented on Client/src/components/game/home.tsx:
```diff
@@ -0,0 +1,13 @@
+import P2Home from "../svg/p2Home";
+import '../../css/home.css';
+
+function home({ xPos, yPos, boxSize, player }: { xPos: number, yPos: number, boxSize: number, player: 'p1' | 'p2' }) {
```
useAppSelector?

vadim-nest replied:
Done

IP1llar commented on Client/src/components/game/leftBar.tsx:
```diff
@@ -0,0 +1,213 @@
+    currentTower: null | TowerType;
+    towers: TowerType[];
+    currentPlayer: string;
+  }
```
useAppSelector

vadim-nest replied:
Working on it..

vadim-nest replied:
Done

IP1llar commented on Client/src/components/game/leftBar.tsx:
```diff
@@ -0,0 +1,213 @@
+  
+  return (
+    <div className="leftBarContainer">
+      <div className="flags">
+        <h3 className="p1Name">{user.username ? user.username : 'You'}</h3>
+        <div className="p1Flag">
+          <FlagSVG
+            playerName="you"
+            playerScore={gameStats.p1Towers.length}
+            playerClass="p1FlagColor"
+            textReverse="text-reverse-nope"
+          />
+        </div>
+        <h1 className="p2Name">Opponent</h1>
+        <div className="p2Flag">
+          <FlagSVG
+            playerName="Isaac"
+            playerScore={gameStats.p2Towers.length}
+            playerClass="p2FlagColor"
+            textReverse="text-reverse-yep"
+          />
+        </div>
+      </div>
```
Need to swap the flags over for player 2

vadim-nest replied:
Looking into it

vadim-nest replied:
Done. Swapping the names instead.

IP1llar commented on Client/src/components/game/leftBar.tsx:
```diff
@@ -0,0 +1,213 @@
+        <h3 className="time-money-text">Time remaining</h3>
+        <h3 className="time-count">{gameStats.timeRemaining}</h3>
+        <h3 className="time-money-text">Money</h3>
+        <h3 className="money-count price">{currentPlayer === 'p1' ? gameStats.p1Coins : gameStats.p2Coins}</h3>
```
Change the colour of money for p2?

vadim-nest replied:
I think we decided not to (the yellow color, the red border for money)

IP1llar commented on Client/src/components/game/maze.tsx:
```diff
@@ -0,0 +1,78 @@
+          {Object.values(minions).map(minion => <Minion key={minion.id} boxSize={boxSize} minion={minion} currentPlayer={currentPlayer}/>)}
+          {!allTilesHidden && towers.map(tower => <Tower key={tower.id} tower={tower} boxSize={boxSize} width={width} height={height}/>)}
+          {maze.map((value: {value: value, classes: string[], path: '' | 'THOUGHTPROCESS' | 'PATH'}, index) => <MazeTile key={index} generated={allTilesHidden} value={value.value as string} path={value.path} classes={value.classes} boxSize={boxSize}/>)}
+        </div>
```
stop prop drilling

vadim-nest replied:
Done

IP1llar commented on Client/src/components/game/maze.tsx:
```diff
@@ -0,0 +1,78 @@
+  
+  useEffect(() => {
+    async function mazeInit(){
+      const mazeTiles = document.getElementsByClassName('mazeTile');
```
Not used

vadim-nest replied:
Deleted

IP1llar commented on Client/src/components/game/mazeTile.tsx:
```diff
@@ -0,0 +1,38 @@
+  return (
+    <>
+      <div onContextMenu={() => setCurrentTileHelper(Number(value))} onClick={() => {
+        dispatch(updateCurrentMinion(null));
+        dispatch(updateCurrentTile(null));
+        dispatch(updateCurrentTower(null));
```
Can these be combined?

IP1llar commented on Client/src/components/game/minion.tsx:
```diff
@@ -0,0 +1,45 @@
+import { minionType, TowerType } from '../../utils/types';
+import { Squirrel, Badger, Hare, Deer, Koala, Bear } from '../svg/animalsSVG';
+
+function Minion({boxSize, minion, currentPlayer}: {
```
Remove props

vadim-nest replied:
Done

IP1llar commented on Client/src/components/game/minion.tsx:
```diff
@@ -0,0 +1,45 @@
+
+  function handleClick() {
+    if (minion.alignment === currentPlayer) {
+      dispatch(updateCurrentTile(null));
+      dispatch(updateCurrentTower(null));
+      dispatch(updateCurrentMinion(minion.id));
```
Combine

IP1llar commented on Client/src/components/game/minion.tsx:
```diff
@@ -0,0 +1,45 @@
+  return (
+    <div onClick={handleClick} onContextMenu={handleContextMenu}  id={`${minion.id}`} className={`minion ${minion.rotation} ${(minion.inTower !== false) && 'minionInTower'}`} style={{fill: `red`, height: `${boxSize + 30}px`, width: `${boxSize + 30}px`, top: `${boxSize*minion.yPos - 15}px`, left: `${boxSize*minion.xPos - 15}px`}}>
+    {
+      minion.type === 'Squirrel' ? <Squirrel currentPlayer={`${minion.alignment}-color`} /> :
+      minion.type === 'Badger' ? <Badger currentPlayer={`${minion.alignment}-color`} /> :
+      minion.type === 'Hare' ? <Hare currentPlayer={`${minion.alignment}-color`} /> :
+      minion.type === 'Deer' ? <Deer currentPlayer={`${minion.alignment}-color`} /> :
+      minion.type === 'Koala' ? <Koala currentPlayer={`${minion.alignment}-color`} /> :
+      minion.type === 'Bear' && <Bear currentPlayer={`${minion.alignment}-color`} />
```
Some sort of hook?

vadim-nest replied:
Done

IP1llar commented on Client/src/components/game/tower.tsx:
```diff
@@ -0,0 +1,52 @@
+  boxSize: number,
+  width: number,
+  height: number,
+} ) {
```
Remove props

vadim-nest replied:
Done

IP1llar commented on Client/src/components/game/tower.tsx:
```diff
@@ -0,0 +1,52 @@
+  
+
+  function handleClick() {
+    dispatch(updateCurrentTower(tower));
+    dispatch(updateCurrentMinion(null));
```
Combine?

IP1llar commented on Client/src/components/game/towerPopup.tsx:
```diff
@@ -0,0 +1,40 @@
+import Visualization from '../sortingLessons/visualization';
+
+
+function TowerPopup ({boxSize, tower, width, height}: {boxSize: number, tower: TowerType, width: number, height: number}) {
```
remove props

IP1llar commented on Client/src/components/game/towerPopup.tsx:
```diff
@@ -0,0 +1,40 @@
+      margin={2} 
+      height={3*boxSize/5} 
+      animations={tower.animations} 
+      sortingAlgo={
+        tower.sortingAlgo === 'bubble' ? bubbleSortVisual :
+        tower.sortingAlgo === 'selection' ? selectionSortVisual :
+        tower.sortingAlgo === 'insertion' ? insertionSortVisual :
+        tower.sortingAlgo === 'merge' ? mergeSortVisual :
+        quickSortVisual
+      } 
```
use a hook

IP1llar commented on Client/src/__tests__/App.test.tsx:
```diff
@@ -0,0 +1,5 @@
+import 'jest';
```
Finish unit tests

IP1llar commented on server/socket.ts:
```diff
@@ -0,0 +1,345 @@
+
+export default function Connect(server: http.Server) {
+
+  let waiting: {socketId: string, roomId: string}[] = [];
+  let ready: {[roomId: string]: string} = {};
+  let readyHost: {[roomId: string]: string} = {};
```
Should these be in the database?

IP1llar commented on server/socket.ts:
```diff
@@ -0,0 +1,345 @@
+    },
+  });
+
+  function clearRoomsBySocket(socket) {
```
Separate game functions from socket things

IP1llar commented on server/controllers/maze.ts:
```diff
@@ -0,0 +1,18 @@
+    try {
+        const {roomId} = req.body;
+        console.log(req.body)
+        const {graph,visited,classes,towers} = mazes[roomId];
```
Check these are valid

IP1llar commented on server/controllers/user.ts:
```diff
@@ -1,22 +1,20 @@
-            const newUser = await User.create({ id: sub, email: email, username: username, avatar:id });
-            const resUser = await User.findOne({email:newUser.email}).populate('avatar');
-            res.status(201).json({ "statusCode": 201, "message": "User created", user: resUser });
+            const username = email.substr(0,email.indexOf('@')).split('.').join(' ');
```
Capitalise each word?

IP1llar commented on README.md:
```diff
@@ -1 +1 @@
-# aMAZEthing
\ No newline at end of file
+# aMAZEthing
```
Should we get this done now or wait for the final week?

josepabellana replied:
Add to taks

josepabellana commented on Client/src/components/game/game.tsx:
```diff
@@ -0,0 +1,298 @@
+import { Graph, value } from '../../utils/graph';
+import { aStar, distanceConstruct, getDirection, vBFS, vDFS, vDijk } from '../../utils/path-finding-algo';
+import { bubbleSortAlgo, insertionSortAlgo, mergeSortAlgo, quickSortAlgo, selectionSortAlgo } from '../../utils/sorting-algo';
+import { uniqueNamesGenerator, Config, names} from 'unique-names-generator'
```
I see two possible hooks here, when requesting visuals and when requesting animations

josepabellana commented on Client/src/components/game/leftBarSmall.tsx:
```diff
@@ -0,0 +1,205 @@
+          <div className={`plus ${toggle}`}>+</div>
+          <div className={`minus ${toggle}`}>-</div>
+        </div>
+      </MediaQuery>
```
I understand this is a work in progress push

MarijaVitkauskaite replied:
yes

josepabellana commented on Client/src/components/game/rightBar.tsx:
```diff
@@ -0,0 +1,136 @@
+    minion.type === 'Deer' ? <Deer currentPlayer={`${minion.alignment}-color`} /> :
+    minion.type === 'Koala' ? <Koala currentPlayer={`${minion.alignment}-color`} /> :
+    minion.type === 'Bear' && <Bear currentPlayer={`${minion.alignment}-color`} />
+  )
```
Can be easily a hook be created to solve this problem?

josepabellana commented on Client/src/components/game/shop.tsx:
```diff
@@ -0,0 +1,63 @@
+  }
+
+  function theRightSVG (animalType: string) {
+    if (animalType === 'Squirrel') return <Squirrel currentPlayer='neutralTower' />
```
the right svg = the right hook

josepabellana commented on Client/src/components/graphVertex.tsx:
```diff
@@ -0,0 +1,27 @@
+    edges:any;
+    width:number;
+    setEnd:any;
+    weightedGraph:boolean;
```
What if instead of having 4 files almost doing the same, just make one that nows what algorithm to use and props needed in every case?

josepabellana commented on Client/src/components/pathfindingLessons/aStar.tsx:
```diff
@@ -0,0 +1,83 @@
+
+  async function aStar(){
+    const aStarVisualpaths = graph.findPath(0, end?end:width*width, "aStar");
+    if(aStarVisualpaths){
```
add slice with the heuristic value

josepabellana commented on Client/src/components/pathfindingLessons/bfsLesson.tsx:
```diff
@@ -0,0 +1,90 @@
+import GraphVertex from "../graphVertex";
+
+function BfsLesson() {
+  const ref: any = useRef(null);
```
delete unused ref

josepabellana commented on Client/src/components/pathfindingLessons/aStar.tsx:
```diff
@@ -0,0 +1,83 @@
+
+
+function AStarLesson() {
+  const ref:any = useRef(null);
```
delete unused ref

josepabellana commented on Client/src/components/pathfindingLessons/dfsLesson.tsx:
```diff
@@ -0,0 +1,82 @@
+
+
+function DfsLesson() {
+  const ref:any = useRef(null);
```
delete unused ref

josepabellana commented on Client/src/components/pathfindingLessons/dfsLesson.tsx:
```diff
@@ -0,0 +1,82 @@
+    }
+  }
+  function delay(time: number) {
+    return new Promise((res) => setTimeout(res, time));
```
export delay function - talk with the team

josepabellana commented on Client/src/components/profile/profile.tsx:
```diff
@@ -0,0 +1,192 @@
+import changeMe from "../../assets/profile/changeMe.png";
+
+function Profile() {
+  const { getAccessTokenSilently, isAuthenticated, user } = useAuth0();
```
are we using auth0 here?

josepabellana commented on Client/src/css/aStar-lesson.css:
```diff
@@ -0,0 +1,33 @@
+  gap: 10px;
+  /* grid-auto-flow: column; */
+  place-items: start;
+  transform: rotate(0deg) scale(0.8, 0.8);
```
create one file for all visualizations

MarijaVitkauskaite committed:
feat: game bottom navbar in progress

IP1llar committed:
Revert "feat: game bottom navbar in progress"

This reverts commit bf3597b5d341bc2bcb20b391830a32067b44f905.

majilaii committed:
fix(an insane hacky way of forcing array refresh lmaooooo path finding): qwe

IP1llar committed:
refactor(visualisation.tsx): set default values for props

josepabellana committed:
style(style formatting, cleaning code): cleaning code

IP1llar committed:
refactor(waitingroom.tsx): remove unused imports

josepabellana committed:
fix(fixing jackie push without pull): thanks jackie

josepabellana committed:
fix(trying to puysh): push

vadim-nest committed:
feat: swap the names for the players

IP1llar committed:
refactor(waitingroom.tsx): combine dispatches into single receiveRoomId action

MarijaVitkauskaite committed:
fix: scroll fix

IP1llar committed:
refactor(waitingroom.tsx): get roomId from useAppSelector

IP1llar committed:
refactor(waitingroom.tsx): remove unused comment

IP1llar committed:
refactor(client socket.ts): remove log

josepabellana committed:
style(cleaning learning code): clean learning

MarijaVitkauskaite committed:
fix: change to useMediaQuery for cleaner code

vadim-nest committed:
refactor: tidy up leftBar.tsx / leftBar.css

josepabellana committed:
style(learning style 2): learning style 2

josepabellana committed:
fix(fix merge problems in learning): fix

MarijaVitkauskaite committed:
fix: update local storage

IP1llar committed:
refactor(game.tsx): gice each animal a movementSpeed

IP1llar committed:
feat(gameover.tsx): implement draw in gameOver

josepabellana committed:
style(put delay function apart): create functionalities file

josepabellana committed:
style(transforming 4 css files into 1): reducing css files

vadim-nest committed:
refactor: useDispatch for rightBar.tsx

josepabellana committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

majilaii committed:
refactor(refactor using usealgo and fixed css bug at game): qwe

IP1llar committed:
fix(store.ts): store overwriting localstorage

IP1llar committed:
refactor(maze.tsx): remove log and unneeded react fragment

IP1llar committed:
refactor(waitingroom.tsx): remove comment

vadim-nest committed:
refactor: a more descriptive name than 'bool' in profile.tsx

vadim-nest committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

vadim-nest committed:
refactor: clean up towerSVG.tsx

josepabellana committed:
style(improve findlearning tsx code): use global variables

IP1llar committed:
refactor(apiservice): usee ApiService type

IP1llar committed:
refactor(sorting-helper-visual.ts): remove comments

josepabellana committed:
style(change var names that were not mutable): change var declaration

josepabellana committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

majilaii committed:
refactor(refactor algos and visual): qw

e

majilaii committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

vadim-nest committed:
refactor: useAppSelector

IP1llar committed:
fix(game.tsx, rightbar.tsx, shop.tsx): fix shop add minion

MarijaVitkauskaite committed:
refactor: change welcome and learning MediaQuery to useMediaQuery

IP1llar committed:
fix(rightbar.tsx): add proper type

vadim-nest committed:
refactor: useAppSelector

MarijaVitkauskaite committed:
refactor: delete leftBarSmall

majilaii committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

josepabellana committed:
style(text changes): path descriptions

josepabellana committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

IP1llar committed:
fix(waitingroom.tsx): fix search

josepabellana committed:
refactor(change comments on path learning): add text

MarijaVitkauskaite committed:
refactor: group game CSS files together

vadim-nest committed:
refactor(create a new hook): create a new hook

vadim-nest committed:
refactor: using the whichAnimalSVG hook

majilaii committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

majilaii committed:
refactor(button logic at sorting): qwe

IP1llar committed:
refactor(waitingroom.tsx): add useEffect logging roomId

vadim-nest committed:
refactor: reuse whicAnimalSVG hook in rightBar.tsx

IP1llar committed:
fix(redux store): set serializable check to false

vadim-nest committed:
refactor: get rid of some of the prop drilling

IP1llar committed:
refactor(game redux): combine finalGameStats and updateGameEnded actions

IP1llar committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

IP1llar committed:
refactor(game.tsx): remove opponentMovement function and just use redux

vadim-nest committed:
refactor: remove not used variables

vadim-nest committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

josepabellana committed:
refactor(deleting fixed margin and padding on titles): fixing css

IP1llar committed:
refactor(game.tsx, hooks.tsx): add pathalgo hook

vadim-nest committed:
refactor: separate some logic in the waiting room into a separate function

majilaii committed:
refactor(refactor visual with types): qwoien

majilaii committed:
Merge branch 'frontend' of https://github.com/IP1llar/aMAZEthing into frontend

majilaii committed:
refactor(deleeted an interface): extra interface not used

qwoei

josepabellana committed:
refactor(testing navbar issues with scrolling): test

vadim-nest committed:
refactor: add some global color variables

IP1llar committed:
fix(learning.tsx): fix scroll bug

josepabellana committed:
refactor(check conflicts): check conflicts

josepabellana committed:
refactor(checking navbar view): check navview

josepabellana committed:
refactor(blocking buttons when pushed): path Learning

IP1llar reviewed (approved):
Good effort
